packages/data-context/src/codegen/spec-options.ts

```diff
@@ -167,36 +128,37 @@ export class SpecGenerator {
           'import { composeStories } from "@storybook/testing-react"',
           `import * as stories from "./${this.parsedPath.name}"`,
         ],
-        mount: (StoryName: string) => `mount(<${StoryName} />)`,
+        stories: csf.stories.map((story) => {
+          const component = story.name.replace(/\s+/, '')
+
+          return {
+            component,
+            mount: `mount(<${component} />)`,
+          }
+        }),
+        fileName: this.getFilename(this.parsedPath.ext),
       },
       vue: {
         imports: [
           'import { mount } from "@cypress/vue"',
           'import { composeStories } from "@storybook/testing-vue3"',
           `import * as stories from "./${this.parsedPath.name}"`,
         ],
-        mount: (StoryName: string) => `mount(${StoryName}())`,
+        stories: csf.stories.map((story) => {
```

Comment:
Will the above `imports` eventually move to a codegen file as well? (as opposed to inlined here?)

Reply:
They could be. Right now we're not looping within ejs.

Reply:
I was mostly wiring it up to the old way of generating specs. I agree, a lot of the options could and should be moved into the template files. We could even have a `vue-component` and a `react-component` template if we didn't want to muddy the templates with too much logic. I'll create an issue for cleaning up the templates.